src/AzureRMTools.ts

```diff
@@ -321,6 +291,145 @@ export class AzureRMTools {
         }
     }
 
+    private logDeploymentTemplateErrors(document: vscode.TextDocument, deploymentTemplate: DeploymentTemplate): void {
+        deploymentTemplate.errors
+            .then((parseErrors: language.Issue[]) => {
+                const diagnostics: vscode.Diagnostic[] = [];
+
+                for (const error of parseErrors) {
+                    diagnostics.push(this.getVSCodeDiagnosticFromIssue(deploymentTemplate, error, vscode.DiagnosticSeverity.Error));
+                }
+
+                for (const warning of deploymentTemplate.warnings) {
+                    diagnostics.push(this.getVSCodeDiagnosticFromIssue(deploymentTemplate, warning, vscode.DiagnosticSeverity.Warning));
+                }
+
+                this._diagnosticsCollection.set(document.uri, diagnostics);
+            })
+            .catch((error: any) => {
+                this.logError("UnhandledError", error);
+            });
+    }
+
+    /**
+     * Update survey settings on disk and return the updated version
+     */
+    private updateSurveySettingsFromOpeningTemplate(): SurveyInfo {
+        const filePath: string = path.join(os.homedir(), ".vscode", "azurerm", "surveyMetadata.json");
+        const settings: SurveySettings = SurveySettings.fromFile(filePath);
+
+        settings.incrementDeploymentTemplatesOpenedOrCreated();
+
+        if (!settings.deploymentTemplateFirstOpenedOrCreatedDateAndTime) {
+            settings.deploymentTemplateFirstOpenedOrCreatedDateAndTime = Date.now();
+        }
+
+        if (settings.showSurveyPrompts === undefined || settings.showSurveyPrompts === null) {
+            settings.showSurveyPrompts = true;
+        }
+
+        settings.toFile(filePath);
+
+        return { settings, filePath };
+    }
+
+    private initializeExtension(surveyInfo: SurveyInfo): void {
```

Comment:
I'm confused - I feel like you copied all of this code without deleting the original code. Am I missing something?

Reply:
Something got messed up.  I've got too many PRs out and getting lots of merges.

Reply:
Fixed